### PR TITLE
CLI: Add TS 4.9 CLI templates

### PR DESCRIPTION
**Pull request by kasperpeulen**
## What I did

* Upgrade prettier to 2.8 for satisfies support
* Upgrade ESLint TS plugin to latest canary
* Making sure we only use TS 4.9 templates when prettier, babel and eslint deps are also in range of supporting satisfies
* Added react, next and svelte tempaltes for TS 4.9
* I didnt add vue2 and vue3 TS templates yet. I'm not sure about the best practices in vue with TS.

### CODE REVIEW

File: code/renderers/svelte/template/cli/js/Button.svelte [hunk 1]
```diff
@@ -22,22 +22,13 @@
...
-  export let onClick = (event) => {
-    dispatch('click', event);
-  };
 </script>
 
 <button
   type="button"
   class={['storybook-button', `storybook-button--${size}`, mode].join(' ')}
   {style}
-  on:click={onClick}
```
Comment (ndelangen) [inline code/renderers/svelte/template/cli/js/Button.svelte:40]: Is this related to this PR?

File: code/renderers/svelte/template/cli/js/Header.stories.js [hunk 2]
```diff
@@ -5,24 +5,10 @@ export default {
...
   tags: ['docsPage'],
-  render: (args) => ({
-    Component: Header,
-    props: args,
-    on: {
-      login: args.onLogin,
-      logout: args.onLogout,
-      createAccount: args.onCreateAccount,
-    },
-  }),
```
Comment (ndelangen) [inline code/renderers/svelte/template/cli/js/Header.stories.js:16]: Is this related to this PR?

**Review by ndelangen — COMMENTED** [review]
looks good, except for some svelte changes, that seem unrelated?

**Review by jonniebigodes — APPROVED** [review]
Aside from Norbert's feedback, all is good on my end.

File: code/renderers/svelte/template/cli/js/Button.svelte (hunk 1 above)
Comment (kasperpeulen) [reply code/renderers/svelte/template/cli/js/Button.svelte:40]: Made a new PR for this

Comment (kasperpeulen) [reply code/renderers/svelte/template/cli/js/Button.svelte:40]: https://github.com/storybookjs/storybook/pull/20007

File: code/renderers/svelte/template/cli/js/Header.stories.js (hunk 2 above)
Comment (kasperpeulen) [reply code/renderers/svelte/template/cli/js/Header.stories.js:16]: Made a new PR for this https://github.com/storybookjs/storybook/pull/20007
